### ISSUE TITLE
After click "Reload" button and go back to "Train" page, the value of Model explored(total) is changed to "5" abnormally.

### ISSUE THREAD

**Issue by v-Hailishi** [dotnet/machinelearning-modelbuilder#2213]
**System Information (please complete the following information):**
Windows OS: Windows-11-Enterprise-21H2
ML.Net Model Builder 2019&2022: 16.13.9.2240801(Main Build)
Microsoft Visual Studio Enterprise: 2019(16.11.17); 2022(17.2.6)
**.Net Framework: 4.7.2**

**Describe the bug**
- On which step of the process did you run into an issue: 
After click "Reload" button and go back to "Train" page, the value of Model explored(total) is changed to "5".

**TestMatrix**
https://testpass.blob.core.windows.net/test-pass-data/wikipedia-detox-250-line-data.tsv 

**To Reproduce**

1. Select Create a new project from the Visual Studio start window;
2. Choose the C# Console App (**.NET Framework**) and .NET Framework 4.7.2 project template;
3. Add model builder by right click on the project;
4. Select Data classification scenario>Local (CPU);
5. On Data page, input the data source, then start training.
6. After training is done, click "Reload" or "Reload All" button on the pop-up dialog, the mbconfig file would be reloaded.
9. Go back to Train page, you will see that the value of Model explored(total) is changed from "19" to "5".

**Expected behavior**
After click "Reload" button and go back to "Train" page, the value of Model explored(total) should be keep with "19".

**Screenshots:**
![image](https://user-images.githubusercontent.com/99375895/183605984-3f68c9bd-3474-4d55-b49c-4f69f73e8e1b.png)

**Additional context:**
The bug is repro on both VS2022&VS2019.

**Comment by luisquintanilla** [comment]
@v-Hailishi how do you get into this state? To confirm, this happens after training is complete? Did you save the mbconfig file?

**Comment by v-Hailishi** [comment]
@luisquintanilla Yes, this happens after training is completed, I trained it on a new mbconfig file, after saved the mbconfig file, the mbconfig file only kept the top 5 running history.
Besides, the bug is also repro for other scenarios and when training on the .NET core project.

**Comment by v-Hailishi** [comment]
The bug is still repro on the latest main build 17.19.0.2455701.
![Image](https://github.com/user-attachments/assets/d052fbdb-93d6-4df6-9e90-fa7d44936b58)
